File: app/src/main/java/sg/partying/lcb/android/config/AppConfig.java
package sg.partying.lcb.android.config;

import com.salton123.config.IConfigProvider;

public class AppConfig {
    public enum AppMode {
        Debug, Alpha, Product
    }

    private static IConfigProvider sConfigProvider;

    public static void init(IConfigProvider configProvider) {
        sConfigProvider = configProvider;
        if (sConfigProvider == null) {
            throw new RuntimeException("you must setProvider first");
        }
        AppMode appMode = AppMode.Product;
        sConfigProvider.beforeInit();
        switch (appMode) {
            case Debug:
                sConfigProvider.initDebug();
                break;
            case Alpha:
                sConfigProvider.initAlpha();
                break;
            default:
                sConfigProvider.initProduct();
                break;
        }
        sConfigProvider.afterInit();
    }
}

File: app/src/main/java/sg/partying/lcb/android/config/AppConfig.kt
package sg.partying.lcb.android.config

import com.salton123.config.AppModeEnum
import com.salton123.config.IConfigProvider
import com.salton123.soulove.CommonClassPath

object AppConfig {
    private var sConfigProvider: IConfigProvider? = null
    fun init(configProvider: IConfigProvider?) {
        sConfigProvider = configProvider
        if (sConfigProvider == null) {
            throw IllegalArgumentException("you must setProvider first")
        }
        val appMode = CommonClassPath.appMode
        sConfigProvider!!.beforeInit()
        when (appMode) {
            AppModeEnum.Debug -> sConfigProvider!!.initDebug()
            AppModeEnum.Alpha -> sConfigProvider!!.initAlpha()
            else -> sConfigProvider!!.initProduct()
        }
        sConfigProvider!!.afterInit()
    }
}
